sdk/monitor/azure-monitor-opentelemetry-exporter/src/test/java/com/azure/monitor/opentelemetry/exporter/AzureMonitorExportersEndToEndTest.java

```diff
@@ -194,42 +198,46 @@ private static void generateLog(OpenTelemetry openTelemetry) {
             .emit();
     }
 
-    private static void validateSpan(TelemetryItem telemetryItem) {
+    private static void validateSpan(TelemetryItem telemetryItem) throws IOException {
         assertThat(telemetryItem.getName()).isEqualTo("RemoteDependency");
         assertThat(telemetryItem.getInstrumentationKey()).isEqualTo(INSTRUMENTATION_KEY);
         assertThat(telemetryItem.getTags()).containsEntry("ai.cloud.role", "unknown_service:java");
         assertThat(telemetryItem.getTags())
             .hasEntrySatisfying("ai.internal.sdkVersion", v -> assertThat(v).contains("otel"));
         assertThat(telemetryItem.getData().getBaseType()).isEqualTo("RemoteDependencyData");
-        RemoteDependencyData actualData = (RemoteDependencyData) telemetryItem.getData().getBaseData();
+
+        // azure-json doesn't deserialize subtypes yet, so need to convert the abstract MonitorDomain to RemoteDependencyData
+        RemoteDependencyData actualData = toRemoteDependencyData(telemetryItem.getData().getBaseData());
```

Comment:
@heyams I pushed this to show an option that avoids relying on the (confusing to me at least) `additionalProperties`

(here's the full commit I pushed: https://github.com/Azure/azure-sdk-for-java/pull/41106/commits/75f6fe2845fb0e22d7a73afdf1d3beedc3723f2c)